kernel/kernbench.py

```diff
@@ -84,20 +84,32 @@ def setUp(self):
         Setting up the env for the kernel building
         """
         smg = SoftwareManager()
-        detected_distro = distro.detect()
+        self.detected_distro = distro.detect()
         deps = ['gcc', 'make', 'automake', 'autoconf', 'time', 'bison', 'flex']
-        if 'Ubuntu' in detected_distro.name:
+        if 'Ubuntu' in self.detected_distro.name:
             deps.extend(['libpopt0', 'libc6', 'libc6-dev', 'libpopt-dev',
                          'libcap-ng0', 'libcap-ng-dev', 'elfutils', 'libelf1',
                          'libnuma-dev', 'libfuse-dev', 'libssl-dev'])
-        elif 'SuSE' in detected_distro.name:
+        elif 'SuSE' in self.detected_distro.name:
             deps.extend(['libpopt0', 'glibc', 'glibc-devel',
                          'popt-devel', 'libcap2', 'libcap-devel',
                          'libcap-ng-devel', 'openssl-devel'])
-        elif detected_distro.name in ['centos', 'fedora', 'rhel']:
+        elif self.detected_distro.name in ['centos', 'fedora', 'rhel']:
             deps.extend(['popt', 'glibc', 'glibc-devel', 'libcap-ng',
                          'libcap', 'libcap-devel', 'elfutils-libelf',
                          'elfutils-libelf-devel', 'openssl-devel'])
+            '''
+            In Rhel Based Distro kernel build is failing as system config file
+            has crypto module . This work around disabled those so it allow to
+            build kernel
+            '''
```

Comment:
I have not faced this issue on a fedora. Are you sure this is applicable for all these distros?

Reply:
@harish-24 sure I did not check in other distros I just going with initial commit so tried only RHEL  let me check and accordingly do changes

Reply:
@harish-24 pushed the changes as right now we tested only in rhel and fedora, if in future anyone try on centos and faces the same issue we can have that patch